eclipse/che-operator
# Automate TLS secrets generation for Kubernetes family infrastructures

Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What this PR does

This PR adds checks into Che Operator reconcile loop for required for Che deployment TLS secrets on Kubernetes infrastructure, and if they are absent, it starts job to generate them.

### Which issues this PR fixes or references

https://github.com/eclipse/che/issues/16546

### How to test

- Via `chectl`:
    - Build operator image, push it
    - Disable TLS secrets check and then run:
```sh
checlt server:start --platform=minikube --installer=operator --self-signed-cert --tls --che-operator-image=<operator-image>
```

- Via `local-debug.sh`:
Copy default CR, and change following fields:
```yaml
spec.server.tlsSupport: true
spec.server.selfSignedCert: true
spec.k8s.ingressDomain: <cluster-domain>
```
